common/src/lib.rs

```diff
@@ -39,20 +36,27 @@ pub struct KeccakInclusionToDataRootProofInput {
 
     pub data_root: [u8; 32],   // already matches
     pub keccak_hash: [u8; 32], // already matches
+    // batch_number and chain_id are passed through to prevent proofs from being replayed
+    pub batch_number: u32,
+    pub chain_id: u64,
```

Comment:
just a nit - do we require these be different types? fine if so, just curious if uniform is OK here.